jgrapht-core/src/test/java/org/jgrapht/alg/TransitiveReductionTest.java

```diff
@@ -0,0 +1,141 @@
+/* ==========================================
+ * JGraphT : a free Java graph-theory library
+ * ==========================================
+ *
+ * Project Info:  http://jgrapht.sourceforge.net/
+ * Project Creator:  Barak Naveh (http://sourceforge.net/users/barak_naveh)
+ *
+ * (C) Copyright 2003-2008, by Barak Naveh and Contributors.
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+/* ----------------------
+ * TransitiveReductionTest.java
+ * ----------------------
+ *
+ * Original Author:   Christophe Thiebaud
+ * Contributor(s):
+ *
+ * Changes
+ * -------
+ * 13-August-2015: Initial revision (CT);
+ *
+ */
+package org.jgrapht.alg;
+
+import java.util.*;
+
+import org.junit.*;
+
+public class TransitiveReductionTest {
```

Comment:
Done in latest commit. I was tempted to protect the reduce method against null pointer exception when a null parameter is passed, but I choose not to, as TransitiveClosure does not.